interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,851 @@
+                        f"{pressure}   {arrhenius} /\n"
+                    )
+
+            elif type(reaction) == ct.ChebyshevReaction:
+                P_atm = [reaction.Pmin / ct.one_atm, reaction.Pmax / ct.one_atm]
+                reaction_string += (
+                    f"TCHEB / {reaction.Tmin}  {reaction.Tmax} /\n"
+                    + f"PCHEB / {P_atm[0]}  {P_atm[1]} /\n"
+                    + f"CHEB / {reaction.nTemperature}  {reaction.nPressure} /\n"
+                )
+                for coeffs in reaction.coeffs:
+                    coeffs_row = " ".join([f"{c:.6e}" for c in coeffs])
+                    reaction_string += f"CHEB / {coeffs_row} /\n"
+
+            # need to trim and print third-body efficiencies, if present
+            if type(reaction) in [
+                ct.ThreeBodyReaction,
+                ct.FalloffReaction,
+                ct.ChemicallyActivatedReaction,
+            ]:
+                # trims efficiencies list
+                reduced_efficiencies = {
+                    s: reaction.efficiencies[s]
+                    for s in reaction.efficiencies
+                    if s in solution.species_names
+                }
+                efficiencies_str = "  ".join(
+                    [f"{s}/ {v:.3f}/" for s, v in reduced_efficiencies.items()]
+                )
+                if efficiencies_str:
+                    reaction_string += "   " + efficiencies_str + "\n"
+
+            if reaction.duplicate:
+                reaction_string += "   DUPLICATE\n"
+
+            mech_file.write(reaction_string)
+
+        mech_file.write("END")
+
+    # write thermo data
+    if not skip_thermo and separate_thermo_file:
+        therm_path = main_path / f"{basename}.therm"
+        with open(therm_path, "w") as thermo_file:
+            thermo_file.write(thermo_data_text(solution.species(), input_type="file"))
+        output_files.append(therm_path)
+
+    # TODO: more careful check for presence of transport data?
+    if not skip_transport and all(sp.transport for sp in solution.species()):
+        trans_path = main_path / f"{basename}.tran"
+        write_transport_data(solution.species(), trans_path)
+        output_files.append(trans_path)
+
+    return output_files
+
+
+def convert_mech(
+    input,
+    output_path="",
+    reorder_reactions=True,
+    skip_thermo=False,
+    separate_thermo_file=False,
+    skip_transport=False,
+):
+
+    if isinstance(input, ct.Solution):
+        solution = input
+    else:
+        solution = ct.Solution(str(input))
+
+    return write(
+        solution,
+        output_path,
+        reorder_reactions,
+        skip_thermo,
+        separate_thermo_file,
+        skip_transport,
+    )
+
+
+def main(argv):
+    longOptions = [
+        "input=",
+        "output_path=",
+        "reorder_reactions=",
+        "skip_thermo",
+        "separate_thermo_file",
+        "skip_transport",
+        "help",
+        "debug",
+        "no-validate",
+    ]
```

Comment:
I think that using `argparse` for the interface would be ideal (see `ctml2yaml` as an example).